Index: src/HW3/BTree.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package HW3;\n\npublic class BTree<E extends Comparable<E>> {\n    private boolean isLeaf;\n    private int height;\n    private Comparable key;\n\n\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/HW3/BTree.java b/src/HW3/BTree.java
--- a/src/HW3/BTree.java	(revision 21a7ed45500d800ed6103a1aa8dd6e3bb4eb7c1b)
+++ b/src/HW3/BTree.java	(date 1701186301383)
@@ -1,6 +1,6 @@
 package HW3;
 
-public class BTree<E extends Comparable<E>> {
+public class BTree<Integer extends Comparable<Integer>> {
     private boolean isLeaf;
     private int height;
     private Comparable key;
